batch/batches/ReachJobCleaner/KAsyncReachJobCleaner.php

```diff
@@ -0,0 +1,156 @@
+ * service type = HUMAN
+ * status pending OR processing
+ * if the due date passed for more than overdueTimePercentage
+ * update status to Error with description: Job Due Date was exceeded in $overdueTimePercentage %
+ *
+ * @package Scheduler
+ * @subpackage ReachJobCleaner
+ */
+require_once('/opt/kaltura/app//alpha/lib/interfaces/IBaseObject.php');
+require_once('/opt/kaltura/app/alpha/lib/interfaces/IIndexable.php');
+require_once('/opt/kaltura/app/alpha/lib/interfaces/IRelatedObject.php');
+
+class KAsyncReachJobCleaner extends KPeriodicWorker
+{
+	const SEVEN_DAYS = 604800;
+	const JOB_DUE_DATE_EXCEEDED = 'Job Due Date was exceeded in ';
+	const MAX_PAGE_SIZE = 500;
+
+	/*
+	* @var KalturaReachClientPlugin
+	*/
+	private $reachClientPlugin = null;
+
+	private $overdueTimePercentage = 100;
+
+	/**
+	 * @param KSchedularTaskConfig $taskConfig
+	 */
+	public function __construct($taskConfig = null)
+	{
+		parent::__construct($taskConfig);
+		$this->reachClientPlugin = KalturaReachClientPlugin::get(self::$kClient);
+		$overdueTimePercentage = $this->getAdditionalParams('overdueTimePercentage');
+		$this->overdueTimePercentage = $overdueTimePercentage ? $overdueTimePercentage : 100;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::REACH_JOB_CLEANER;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::run()
+	*/
+	public function run($jobs = null)
+	{
+		KalturaLog::debug('Getting all the entry vendor task items');
+		$filter = new KalturaEntryVendorTaskFilter();
+		$filter->statusIn = KalturaEntryVendorTaskStatus::PENDING . ',' . KalturaEntryVendorTaskStatus::PROCESSING;
+		$filter->updatedAtLessThanOrEqual = time();
+		$filter->updatedAtGreaterThanOrEqual = time() -  (VendorServiceTurnAroundTime::TEN_DAYS + VendorServiceTurnAroundTime::FOUR_DAYS);
+		$filter->orderBy = KalturaEntryVendorTaskOrderBy::UPDATED_AT_ASC;
+
+		$pager = new KalturaFilterPager();
+		$pager->pageSize = self::MAX_PAGE_SIZE;
+		$pager->pageIndex = 1;
+
+		$lastUpdatedAt = 0;
+		$totalCount = 0;
+		do {
+			if ($lastUpdatedAt)
+			{
+				$filter->updatedAtGreaterThanOrEqual = $lastUpdatedAt;
+			}
+			try
+			{
+				$entryVendorTaskList = $this->reachClientPlugin->entryVendorTask->listAction($filter, $pager);
+				$tasksCount = count($entryVendorTaskList->objects);
+			}
+			catch (Exception $e)
+			{
+				KalturaLog::debug("Couldn't list entry Vendor Tasks " . $e->getMessage());
+				return;
+			}
+
+			$this->updateStuckJobs($entryVendorTaskList);
+
+			$totalCount += $tasksCount;
+			KalturaLog::debug("Handled More tasks - $tasksCount totalCount - " . $totalCount);
+			if ($tasksCount)
+			{
+				$lastObject = end($entryVendorTaskList->objects);
+				$lastUpdatedAt = $lastObject->updatedAt + 1;
+			}
+
+			unset($entryVendorTaskList);
+			if (function_exists('gc_collect_cycles')) // php 5.3 and above
+			{
+				gc_collect_cycles();
+			}
+		} while ($pager->pageSize == $tasksCount);

```

Comment:
Instead use tasksCount>0, since it is possible that 1 job was deleted or moved, while we asked for the second page from the first page, and we will get 499 jobs/.